common/src/api/external/mod.rs

```diff
@@ -919,6 +919,8 @@ pub enum DiskState {
     Creating,
     /// Disk is ready but detached from any Instance
     Detached,
+    /// Disk is undergoing maintenance
```

Comment:
To summarize the chat: I'd like first change this label from `Maintenance` to something more specific, like `Taking a Snapshot`, and would like to decouple the internal `DiskState` from the view the user has. The user should only see `Busy` with some `String` reason, so their code can match on `DiskState::Busy(_)` instead of requiring an update every time we add a new maintenance-like operation like this (where the disk needs to be attached to the Pantry).

I'd love to do this in a follow up PR though, this one's been open for a long time and keeps accumulating merge conflicts.

Reply:
Sounds good. If we're aligned on long-term intent, I'm happy to merge this earlier and do follow-up work in separate PRs.